sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusSessionsEndToEndTests.cs

```diff
@@ -1106,6 +1146,41 @@ public class TestReceiveFromFunction
             }
         }
 
+        public class TestCustomSessionInitializingHandler
+        {
+            public static async Task RunAsync(
+                [ServiceBusTrigger(FirstQueueNameKey, IsSessionsEnabled = true)]
+                ServiceBusReceivedMessage message,
+                ServiceBusSessionMessageActions sessionActions)
+            {
+                await sessionActions.CompleteMessageAsync(message);
+            }
+
+            public static Task SessionInitializingHandler(ProcessSessionEventArgs arg)
+            {
+                _waitHandle1.Set();
```

Comment:
I think the issue is that this effectively ends the test before the message is completed, and we have validation that runs at the end of each test to ensure all messages have been drained from the queue. I think we can probably just combine these two tests into one test and set the waitHandle in the closing handler. We would need to set some other private field from the initializing handler, that we can check in the function body.

Reply:
Turned it into one test now.